app/controllers/samples_controller.rb

```diff
@@ -124,40 +124,111 @@ def index
   end
 
   def index_v2
-    only_library = ActiveModel::Type::Boolean.new.cast(params[:onlyLibrary])
-    exclude_library = ActiveModel::Type::Boolean.new.cast(params[:excludeLibrary])
+    domain = params[:domain]
```

Comment:
I'm assuming some of these controller changes are implementing Charles's suggestions?

Reply:
+ some extra improvements.